### PR TITLE
Update Core dependency to 3.0.0 (non snapshot) version

### PR DESCRIPTION
**Pull request by kevinlind**
## Description

Updating dependencies to non-snapshot versions of 3.0.0.

- Sets Core and Identity dependencies in edgeidentity module to non-snapshot versions
- Updates app module to use non-snapshot version of Core. However, as Edge Identity is released before Edge and Consent, the test app must still reference the 3.0.0-SNAPSHOT releases of these extensions.

Note, as Core 3.0.0 is not yet released, the CI build will fail.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/app/build.gradle.kts [hunk 1]
```diff
@@ -69,18 +69,13 @@ dependencies {
...
-        exclude(group = "com.adobe.marketing.mobile", module = "core")
-    }
+    implementation("com.adobe.marketing.mobile:core:$mavenCoreVersion")
+    implementation("com.adobe.marketing.mobile:identity:3.0.0")
     implementation("com.adobe.marketing.mobile:edgeconsent:3.0.0-SNAPSHOT") {
         exclude(group = "com.adobe.marketing.mobile", module = "edge")
-        exclude(group = "com.adobe.marketing.mobile", module = "core")
-
     }
     implementation("com.adobe.marketing.mobile:assurance:2.+")
```
Comment (cacheung) [inline code/app/build.gradle.kts:77]: Assurance should use 3.0.0-SNAPSHOT

File: code/edgeidentity/build.gradle.kts [hunk 2]
```diff
@@ -41,7 +39,7 @@ dependencies {
     // ANDROIDX_TEST_EXT_JUNIT, ESPRESSO_CORE
 
     androidTestImplementation ("com.fasterxml.jackson.core:jackson-databind:2.12.7")
-    androidTestImplementation("com.adobe.marketing.mobile:identity:$functionalTestIdentityVersion-SNAPSHOT")
+    androidTestImplementation("com.adobe.marketing.mobile:identity:$functionalTestIdentityVersion")
```
Comment (cacheung) [inline code/edgeidentity/build.gradle.kts:42]: Do we still need to exclude core here?

**Review by cacheung — COMMENTED** [review]
Can we also include the readme change in this PR?
https://github.com/kevinlind/aepsdk-edgeidentity-android/tree/staging-v3-dep-updates

To open and run the project, open the code/settings.gradle.kt file in Android Studio.

File: code/edgeidentity/build.gradle.kts (hunk 2 above)
Comment (kevinlind) [reply code/edgeidentity/build.gradle.kts:42]: No, identity is bundled with Core and they are set to the same version.

**Comment by emdobrin** [comment]
could we also include the readme/documentation getting started with the app doc changes with this pr?

File: code/edgeidentity/build.gradle.kts (hunk 2 above)
Comment (emdobrin) [reply code/edgeidentity/build.gradle.kts:42]: let's update he lines below 👇

File: code/app/build.gradle.kts [hunk 3]
```diff
@@ -69,18 +69,13 @@ dependencies {
...
-    }
+    implementation("com.adobe.marketing.mobile:core:$mavenCoreVersion")
+    implementation("com.adobe.marketing.mobile:identity:3.0.0")
     implementation("com.adobe.marketing.mobile:edgeconsent:3.0.0-SNAPSHOT") {
         exclude(group = "com.adobe.marketing.mobile", module = "edge")
-        exclude(group = "com.adobe.marketing.mobile", module = "core")
-
     }
-    implementation("com.adobe.marketing.mobile:assurance:2.+")
+    implementation("com.adobe.marketing.mobile:assurance:3.0.0")
```
Comment (emdobrin) [inline code/app/build.gradle.kts:77]: if we change this to 3.0.0-SNAPSHOT we can merge into staging directly, just saying 👯

**Review by emdobrin — APPROVED** [review]

**Pull request merged**